src/tools/wasm-interp.cc

```diff
@@ -110,6 +117,13 @@ static void ParseOptions(int argc, char** argv) {
   parser.AddOption(
       'd', "dir", "DIR", "Pass the given directory the the WASI runtime",
       [](const std::string& argument) { s_wasi_dirs.push_back(argument); });
+  parser.AddOption(
+      'm', "module", "[(@|$)NAME:]FILE",
```

Comment:
I wonder if we can come up with better name for this parameter?     How about something like `--preload`?

Also perhaps we could just get away without adding a new option here at all and just treat these module like main binary we are running. i.e could we just allow wasm-interp to take any number of binaries as arguments rather than the current limit of just one?

Reply:
i wanted module to be used for the import/exprot because thats what its called in the docs of WebAssembly. `--preload` would not be to far fetched, but I still think `--module` (or just `--mod`) would still be best. (You can choose, just wanted to give my reasoning).

Sadly if we want it to interoperate with `--wasi` (and as long as we don't have a glue for native libraries) a new option is a must.
I was thinking about multiple different ways, but the OptionParser does not really like them. I cannot specify multiple modules, then a main one, and then optional parameters without *some* new option to mark the main module. I guessed `-m` for conveniance á la '-L"path" -llibname' would be best and in the end just stuck to the `-m`/`--module` to be more like the other options available.

I was thinking about something to glue native libraries and just have `--wasi` be an indicator which native library to give the optional arguments to.


references:
https://webassembly.github.io/spec/core/valid/modules.html#imports
https://webassembly.github.io/spec/core/text/modules.html#imports
https://webassembly.github.io/spec/core/syntax/modules.html#imports
https://webassembly.github.io/spec/core/binary/modules.html#import-section

Reply:
Still a new option, but if I waht you want, this would be better suited:
we could have an `-a, --arg` parameter right before the arguments. I just don't know how to make it optional and still have a list of arguments following it.

Reply:
I know now why I did this, only the main module has access to `--wasi` the other ones don't.
I first need to figure out how to provide wasi support to all of them, then how to do the `--arg` thingy and then I don't mind.

Reply:
Speaking of `-L`.. I wonder if we could have wasi-interp automatically looks for wasm files in certain locations and load them on demand to satisfy imports?   

We could even do something fancy like JS import maps: https://github.com/WICG/import-maps

But all that is farther down the line after we land this initial support.